src/main/java/org/elasticsearch/cluster/ClusterState.java

```diff
@@ -642,5 +674,84 @@ public static ClusterState readFrom(StreamInput in, @Nullable DiscoveryNode loca
             }
             return builder.build();
         }
+
+    }
+
+    public Diff diffs(ClusterState previousState) {
+        return new ClusterStateDiff(previousState, this);
+    }
+
+
+    private class ClusterStateDiff implements Diff {
+
+        private final long version;
```

Comment:
can we maybe name this with something that indicates its the version of the after ClusterState? `afterVersion` sounds bad. If we use `to` and `from` in the ctor we can have toVersion, fromUuid, toUuid etc :)